ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/attestation/AggregatingAttestationPool.java

```diff
@@ -59,6 +62,7 @@ public AggregatingAttestationPool(
   public synchronized void add(final ValidateableAttestation attestation) {
     final AttestationData attestationData = attestation.getAttestation().getData();
     final Bytes32 dataRoot = attestationData.hash_tree_root();
+    size.getAndIncrement();
     attestationGroupByDataHash
         .computeIfAbsent(dataRoot, key -> new MatchingDataAttestationGroup(attestationData))
         .add(attestation);
```

Comment:
The attestation winds up being added to a set so it deduplicates - we'd need to have the `add` call return a flag to indicate if the attestation was actually added or not.